src/SmartComponents/DriftPage/FilterDropDown/FilterDropDown.js

```diff
@@ -9,44 +9,43 @@ class FilterDropDown extends Component {
     constructor(props) {
         super(props);
         this.onToggle = this.onToggle.bind(this);
-        this.getFilterbyState = this.getFilterByState.bind(this);
         this.createDropdownItem = this.createDropdownItem.bind(this);
     }
 
     onToggle() {
         this.props.toggleDropDown();
     }
 
-    getFilterByState(event, filter) {
-        this.props.changeFilter(filter);
-    }
-
-    createDropdownItem(display, value) {
-        let dropdownItem = <DropdownItem key={ value } component="button" onClick={ event =>
-            this.getFilterByState(event, value)
-        }>
-            { display }
-        </DropdownItem>;
+    createDropdownItem(idValue, display, value, checkedValue) {
+        let dropdownItem = <Checkbox id={ idValue } aria-label={ idValue } label={ display } isChecked={ checkedValue } onChange={ () =>
+            this.props.addStateFilter(value)
+        } />;
 
         return dropdownItem;
     }
 
     render() {
+        const { same, different, incomplete, stateFilters } = this.props;
         const dropdownItems = [
-            this.createDropdownItem('All', 'All'),
-            this.createDropdownItem('Same', 'SAME'),
-            this.createDropdownItem('Different', 'DIFFERENT'),
-            this.createDropdownItem('Incomplete Data', 'INCOMPLETE_DATA')
+            this.createDropdownItem('same-check', 'Same', 'SAME', same),
+            this.createDropdownItem('different-check', 'Different', 'DIFFERENT', different),
+            this.createDropdownItem('incomplete-check', 'Incomplete Data', 'INCOMPLETE_DATA', incomplete)
         ];
 
+        let selectedViews = '';
+        for (let i = 0; i < stateFilters.length; i++) {
+            selectedViews += (stateFilters[i].charAt(0).toUpperCase() + stateFilters[i].slice(1).toLowerCase()).replace(/_/g, ' ');
```

Comment:
Given this and the above, I feel like this might be the time to consider a re-factor to a data structure that holds the different filters. Possibly defined in the state?